### DISCUSSION TITLE
learning rate (lr) not doing much for yourtts, but does for vctk vits

### DISCUSSION THREAD

**Discussion by rtruszkowski** [coqui-ai/TTS#2813] [General Q&A] [answered]
I've been fine-tuning two models: `tts_models--multilingual--multi-dataset--your_tts` and `tts_models--en--vctk--vits`.

I noticed that different learning rates (`lr`) change my loss graphs a lot when fine-tuning `vctk--vits`, and do absolutely nothing for `your_tts`. When I fine-tune `your_tts`, the `lr` can be `1e-3` or `1e-8` and the resulting loss graphs are the same. This doesn't make any sense in my brain.

When using the exact same code, and only swapping the `restore_path` between two models and modifying `lr` for `VitsArgs`, `vctk--vits` loss functions change with `lr`, but `yourtts` loss functions do not. 

Is that... expected?

my VitsConfig:
```

config = VitsConfig(
    characters=CharactersConfig(
        characters_class="TTS.tts.models.vits.VitsCharacters",
        pad="_",
        eos="&",
        bos="*",
        blank=None,
        characters="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz\u00af\u00b7\u00df\u00e0\u00e1\u00e2\u00e3\u00e4\u00e6\u00e7\u00e8\u00e9\u00ea\u00eb\u00ec\u00ed\u00ee\u00ef\u00f1\u00f2\u00f3\u00f4\u00f5\u00f6\u00f9\u00fa\u00fb\u00fc\u00ff\u0101\u0105\u0107\u0113\u0119\u011b\u012b\u0131\u0142\u0144\u014d\u0151\u0153\u015b\u016b\u0171\u017a\u017c\u01ce\u01d0\u01d2\u01d4\u0430\u0431\u0432\u0433\u0434\u0435\u0436\u0437\u0438\u0439\u043a\u043b\u043c\u043d\u043e\u043f\u0440\u0441\u0442\u0443\u0444\u0445\u0446\u0447\u0448\u0449\u044a\u044b\u044c\u044d\u044e\u044f\u0451\u0454\u0456\u0457\u0491\u2013!'(),-.:;? ",
        punctuations="!'(),-.:;? ",
        phonemes="",
        is_unique=True,
        is_sorted=True,
    ),
    target_loss="loss_1",
    add_blank=True,
    audio=audio_config,
    batch_size=BATCH_SIZE,
    compute_input_seq_cache=True,
    cudnn_benchmark=True,
    dashboard_logger="tensorboard",
    datasets=DATASETS_CONFIG_LIST,
    eval_batch_size=BATCH_SIZE,
    log_model_step=500,
    logger_uri=None,

    lr=1e-05,

    max_audio_len=SAMPLE_RATE * MAX_AUDIO_LEN_IN_SECONDS,
    max_text_len=475,
    mixed_precision=False,
    model_args=model_args,
    num_eval_loader_workers=0,
    num_loader_workers=0,
    output_path=output_path,
    phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    phoneme_language="en-us",
    phonemizer="espeak",
    plot_step=100,
    print_eval=True,
    print_step=50,
    project_name="YourTTS",
    run_description="""YourTTS VCTK training test""",
    run_eval=True,
    run_name=RUN_NAME,
    save_all_best=False,
    save_checkpoints=True,
    save_n_checkpoints=3,
    save_step=250,
    scheduler_after_epoch=True,
    speaker_encoder_loss_alpha=9.0,
    start_by_longest=False,
    test_delay_epochs=-1,
    text_cleaner="multilingual_cleaners",
    use_language_weighted_sampler=True,
    use_speaker_weighted_sampler=True,
    use_phonemes=False,
    # use_weighted_sampler=True,
    # weighted_sampler_attrs={"speaker_name": 1.0},
    # weighted_sampler_multipliers={"VCTK_new_voice":27},
    eval_split_size=0.0625,
    test_sentences=[
        [
            "It took me quite a long time to develop a voice, and now that I have it I'm not going to be silent.",
            "VCTK_new_voice",
            None,
            "en",
        ],
        [
            "My name is Rafal. I did not record this. This voice was computer generated.",
            "VCTK_new_voice",
            None,
            "en",
        ],
        [
            "I'm sorry Dave. I'm afraid I can't do that.",
            "VCTK_new_voice",
            None,
            "en",
        ],
        [
            "This cake is great. It's so delicious and moist.",
            "VCTK_new_voice",
            None,
            "en",
        ],
        [
            "Rafal Truszkowski is an amazing software engineer!",
            "VCTK_new_voice",
            None,
            "en",
        ],
    ],
)
```

Four different learning rates, from `1e-4` to `1e-7`
<img width="1004" alt="Screenshot 2023-07-28 at 4 59 58 PM" src="https://github.com/coqui-ai/TTS/assets/107276407/d9fb5887-b6fc-4f45-b92b-bfe45a5321f8">

**Comment by lexkoro** [comment, marked as answer]
Because you are most likely not setting the learning rate correctly.

It's
```
    lr_gen=2e-4,
    lr_disc=2e-4,
```

see https://github.com/coqui-ai/TTS/blob/b739326503b6077c8662c0244b461ca75294384e/TTS/tts/configs/vits_config.py#L25C9-L25C16

**Reply by rtruszkowski** [reply]
huh... I should have checked the src directly, you're right! I was using `lr` because I relied on this page: https://tts.readthedocs.io/en/latest/models/vits.html#vitsconfig 🤔 which has a `lr` parameter.

**Reply by rtruszkowski** [reply]
Ah... but it's not defined further down the page =( That's my bad haha. Thank you!

**Comment by lexkoro** [comment]
Yeah it's inherited all the way up from 
https://github.com/coqui-ai/Trainer/blob/2a5a13ed57a15af180c8af33b171f2c17c230ee8/trainer/trainer.py#L65 😅

**Reply by rtruszkowski** [reply]
Cool. Thanks so much! I was going crazy... lol
